File: BankKycSystemdotnet/BankKycSystem/Entities/AuditLog.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vaultxback.Models
{
    public class AuditLog
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        public string? Action { get; set; }

        public string? PerformedBy { get; set; }

        public DateTime? Timestamp { get; set; }

        public string? TargetEntity { get; set; }

        public string? Details { get; set; }
    }
}

File: bank-kyc-system/src/main/java/com/bank/project/entity/AuditLog.java
package com.bank.project.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String action;          
    @Column
    private String performedBy;     
    @Column
    private LocalDateTime timestamp;
    @Column
    private String targetEntity;    
    @Column
    private String details;         
}
